### PR TITLE
feat: Automatic assignment Tracking

### PR DESCRIPTION
**Pull request by tyiuhc**
### Summary

Added auto-assignment tracking feature

### Checklist

* [X] Does your PR title have the correct [title format](https://github.com/amplitude/experiment-go-server/blob/main/CONTRIBUTING.md#pr-commit-title-conventions)?
* Does your PR have a breaking change?:  No

### CODE REVIEW

File: internal/cache/cache.go [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+type Item struct {
+	Key       string
+	Value     interface{}
+	ExpiresAt time.Time
+}
+
+type Cache struct {
+	Capacity  int
+	TTL       int
```
Comment (bgiori) [inline internal/cache/cache.go:16]: In go we should use `time.Duration` for any duration related fields like TTL.

File: internal/cache/cache.go [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+	cacheMap  map[string]*list.Element
+	cacheList *list.List
+}
+
+func NewCache(capacity int, ttlMillis int) *Cache {
```
Comment (bgiori) [inline internal/cache/cache.go:21]: time.Duration

File: pkg/experiment/local/assignment_service.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import (
+	"fmt"
+	"github.com/amplitude/analytics-go/amplitude"
+)
+
+const DayMillis = 24 * 60 * 60 * 1000
+const FlagTypeMutualExclusionGroup = "mutual-exclusion-group"
+const FlagTypeHoldoutGroup = "mutual-holdout-group"
+
+type AssignmentService struct {
```
Comment (bgiori) [inline pkg/experiment/local/assignment_service.go:12]: These should all be private

File: pkg/experiment/local/assignment_service.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	Amplitude *amplitude.Client
+	Filter    *assignmentFilter
```
Comment (bgiori) [inline pkg/experiment/local/assignment_service.go:14]: These can also probably be private.

**Review by bgiori — CHANGES_REQUESTED** [review]
Brilliant, couple minor changes, but looks great.

File: pkg/experiment/local/assignment_service.go (hunk 3 above)
Comment (tyiuhc) [reply pkg/experiment/local/assignment_service.go:12]: DayMillis is also used in AssignmentFilter, FlagTypeMutualExclusionGroup and FlagTypeHoldoutGroup are both also used in local client

**Review by bgiori — APPROVED** [review]

**Pull request merged**